custom_components/unifiprotect/__init__.py

```diff
@@ -93,31 +87,23 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if not protect_data.last_update_success:
         raise ConfigEntryNotReady
 
-    update_listener = entry.add_update_listener(_async_options_updated)
-    data = hass.data[DOMAIN][entry.entry_id] = {
+    hass.data[DOMAIN][entry.entry_id] = {
         "protect_data": protect_data,
         "upv": protectserver,
         "snapshot_direct": entry.options.get(CONF_SNAPSHOT_DIRECT, False),
         "server_info": nvr_info,
-        "update_listener": update_listener,
         "disable_stream": entry.options.get(CONF_DISABLE_RTSP, False),
         "doorbell_text": entry.options.get(CONF_DOORBELL_TEXT, None),
     }
 
-    async def _async_stop_protect_data(event):
-        """Stop updates."""
-        await protect_data.async_stop()
-
-    data["stop_event_listener"] = hass.bus.async_listen_once(
-        EVENT_HOMEASSISTANT_STOP, _async_stop_protect_data
-    )
-
     await _async_get_or_create_nvr_device_in_registry(hass, entry, nvr_info)
 
-    for platform in UNIFI_PROTECT_PLATFORMS:
-        hass.async_create_task(
-            hass.config_entries.async_forward_entry_setup(entry, platform)
-        )
+    hass.config_entries.async_setup_platforms(entry, UNIFI_PROTECT_PLATFORMS)
+
+    entry.async_on_unload(entry.add_update_listener(_async_options_updated))
+    entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, protect_data.async_stop)
```

Comment:
When the callback runs from `async_listen_once` it will call `async_stop` with the stop event as the first argument

protect_data.async_stop(event)

If you change the signature of `async_stop` to something like `async_stop(self, *args)` it will not throw the error anymore.

Reply:
Thanks, this did the trick.